docs/game-mechanics/behaviors/basic-attack.rst

```diff
@@ -1,39 +1,87 @@
-Basic Attack (1)
-================
+`Basic Attack (1) <https://github.com/DarkflameUniverse/DarkflameServer/blob/main/dGame/dBehaviors/BasicAttackBehavior.cpp>`_
+=============================================================================================================================
 
-This behavior is used to deal damage to the current target.
+This behavior is used to deal damage to a target.
 
-Parameters
-----------
+`Parameters <https://github.com/DarkflameUniverse/DarkflameServer/blob/main/dGame/dBehaviors/BasicAttackBehavior.cpp#L142>`_
+----------------------------------------------------------------------------------------------------------------------------
 
 .. list-table ::
-   :widths: 15
+   :widths: 15 30
    :header-rows: 1
 
    * - Name
+     - Description
    * - dir_angle_xz
+     - The direction to modify the attackers angle by.
    * - dir_angle_y
+     - The direction to modify the attackers angle by.
    * - dir_force
+     - The amount of force to apply towards angle specified in dir_angle_xz or dir_angle_y.
    * - dont_apply_immune
+     - Whether or not to apply immunity to the caster of the behavior.
    * - max damage
+     - The maximum amount of damage to be dealt.
    * - min damage
+     - The minimum amount of damage to be dealt.
    * - on_fail_armor
+     - The behavior to use on failure due to armor.
    * - on_fail_blocked
+     - The behavior to use on failure due to the attack being blocked.
    * - on_fail_immune
+     - The behavior to use on failure due to the target being immune.
    * - on_success
+     - The behavior to use on success of the attack.
    * - radius
+     - The radius of the attack.
    * - use_caster_velocity
+     - Whether or not to use the casters velocity.
    * - velocity_multiplier
+     - Value to multiply velocity by when attacking.
+
+DarkFlame Universe Parameter Notes
+----------------------------------
+
+As of February 11, 2022, only the following parameters are used:
+
+- max damage
+- min damage
+- on_success
+
+The following parameters are not attached to any behavior trees or skills from live:
+
+- dir_angle_xz
+- dir_angle_y
+- dir_force
+- radius
+- use_caster_velocity
+- velocity_multiplier
+
+It is unknown whether the following need to be used:
+
+- on_fail_armor
+- on_fail_blocked
+- on_fail_immune
+- dont_apply_immune
 
 BitStream Serialization
 -----------------------
 
-align to byte boundary (don’t ask me why, this (and the “padding” below) is completely pointless)
+Align to byte boundary.
+
+| **[u16]** - Required BitStream Padding.
+| **[bit]** - True if the attack was blocked, false otherwise.
+| **[bit]** - True if the the target is immune, false otherwise.
+| **[bit]** - True if the attack was successful, false otherwise.
+| **[u32]** - Required BitStream Padding.
```

Comment:
I find it unlikely that these are actually padding with no meaning at all. Is there any proof somewhere that it isn't just something missing in DLU?

Reply:
@Jettford has mentioned in Discord that the padding is all zeros.  as for its meaning I have no idea but if it needs to be done there has to be a meaning to it right?  After all why do work that is pointless?

Reply:
exactly my thinking

Reply:
The client literally logs it as `BasicAttackBehavior--> Padding is %d`, it is padding.

Reply:
both of those?

Reply:
both?

Reply:
Line 72 and 76

Reply:
Hmm, I hadn't noticed 72, I will take a look in a bit

Reply:
I recall asking about these in discord as well - I recall asking what unknown is and being told it was just all zeros.  If that isn't exactly what padding is then my mistake.

Reply:
Generally, padding is bytes/bits that are not assigned and thus can be arbitrary but are usually 0 for well behaved systems. I would call the bits before the byte alignment "padding". As Jett mentioned and the docs before had `"padding"` in 72, at least one of those is probably actual padding.

But because it's so unusual (which LU has a lot of as you know by now), it may be good to take this opportunity and write down why someone on discord might be correct in replying that.

It's a bootstraping thing in some sense: If the person on discord says it's padding because the docs do and the docs say it's padding because someone in discord said it, that's not very helpful. OTOH if we say the client logs padding when it reaches this and that's confirmed by RE, that's a completely different story and can be written down.

Thank you very much for putting all this effort in btw, I'm not trying to be pedantic, just making sure we discuss this now, so in the future people can just look at the docs and understand why they're the way they are.